### DISCUSSION TITLE
Push to talk key binding

### DISCUSSION THREAD

**Discussion by make-github-pseudonymous-again** [i3/i3#4706] [Q&A]
Has anybody implemented something similar to https://glow.li/posts/global-push-to-talk/? Is there a flag similar to `--release` that would only execute the bound command on initial key press to avoid the repeat behavior of i3 when maintaining a key pressed?

Without such a solution, an inordinate amount of exec calls are fired. Although in this particular example unmuting the mic constitutes an idempotent operation, one may wonder if avoiding these unnecessary calls is possible.

**Comment by Airblader** [comment]
> to avoid the repeat behavior of i3

That's not really "i3 behavior", it's the X server that fires a key press event that often.

You can always use xset to disable key repetition for specific keys.

**Reply by orestisfl** [reply]
IIRC, it's possible to distinguish between repeated key presses. In which case, i3 could implement such behaviour (ignore key press for command until release).

**Reply by make-github-pseudonymous-again** [reply]
Isn't there such a behavior in the browser, independently of X? `keydown` and `keyup`?

**Reply by Airblader** [reply]
I never said it can't be tracked and implemented. I said that key repetition is not caused by i3.

**Reply by make-github-pseudonymous-again** [reply]
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/repeat
